common/djangoapps/microsite_configuration/backends/database.py

```diff
@@ -0,0 +1,202 @@
+        return config.get(val_name, default)
+
+    def get_all_orgs(self):
+        """
+        This returns a set of orgs that are considered within a microsite. This can be used,
+        for example, to do filtering
+        """
+
+        # This should be cacheable (via memcache to keep consistent across a cluster)
+        # I believe this is called on the dashboard and catalog pages, so it'd be good to optimize
+        return set(MicrositeOrgMapping.objects.all().values_list('org', flat=True))
+
+    def _set_microsite_config_from_obj(self, subdomain, domain, microsite_object):
+        """
+        Helper internal method to actually find the microsite configuration
+        """
+        config = microsite_object.values
+        config['subdomain'] = subdomain
+        config['site_domain'] = domain
+        config['microsite_config_key'] = microsite_object.key
+
+        # we take the list of ORGs associated with this microsite from the database mapping
+        # tables. NOTE, for now, we assume one ORG per microsite
+        orgs = microsite_object.get_orgs()
+
+        # we must have at least one ORG defined
+        if not orgs:
+            raise Exception(
+                'Configuration error. Microsite {key} does not have any ORGs mapped to it!'.format(
+                    key=microsite_object.key
+                )
+            )
+
+        # just take the first one for now, we'll have to change the upstream logic to allow
+        # for more than one ORG binding
+        config['course_org_filter'] = orgs[0]
+        self.current_request_configuration.data = config
+
+
+class DatabaseMicrositeTemplateBackend(BaseMicrositeTemplateBackend):
+    """
+    Specialized class to pull templates from the database.
+    This Backend would allow us to save templates in DB and pull
+    them from there when required for a specific microsite.
+
+    E.g. we have setup a microsite for subdomain `mua.edx.org` and
+    We have a DB entry like this in table created by MicrositeTemplate model.
+
+    microsite = Key for microsite(mua.edx.org)
+    template_uri = about.html
+    template = <html><body>Template from DB</body></html>
+
+    While using DatabaseMicrositeTemplateBackend any request coming from mua.edx.org/about.html
+    would get about.html template from DB and response would the value of `template` column.
+    """
+    def get_template_path(self, relative_path, **kwargs):
+        return relative_path
+
+    def get_template(self, uri):
+        """
+        Override of the base class for us to look into the
+        database tables for a template definition, if we can't find
+        one we'll return None which means "use default means" (aka filesystem)
+        """
+        cache_key = "template_cache." + microsite_get_value('microsite_config_key') + '.' + uri
+        template_text = cache.get(cache_key)  # pylint: disable=maybe-no-member
+
+        if not template_text:
+            # Don't use the cache
```

Comment:
This seems like an odd comment -- we have already checked the cache above and found nothing -- so this operation does indeed use the cache -- let's clarify what is going on here.